src/server/publicApi.js

```diff
@@ -47,22 +47,32 @@ const runPublicApi = (gameServer) => {
 };
 
 const authMiddleware = (gameServer) => async (ctx, next) => {
-  const credentials = auth(ctx);
-  const game = await gameServer.db.fetch(ctx.params.matchID, {
-    metadata: true,
-  });
-  const metadata = game.metadata;
-  if (
-    credentials &&
-    credentials.name &&
-    metadata &&
-    metadata.players &&
-    credentials.pass === metadata.players[credentials.name].credentials
-  ) {
-    return await next();
-  } else {
-    ctx.throw(403);
+  try {
+    const credentials = auth(ctx);
+    const game = await gameServer.db.fetch(ctx.params.matchID, {
+      metadata: true,
+    });
+    const metadata = game.metadata;
+
+    if (
+      credentials.name === SPECTATOR &&
+      credentials.pass === metadata.setupData.spectatorCredential
+    ) {
+      return await next();
+    }
+
+    if (credentials.pass === metadata.players[credentials.name].credentials) {
+      return await next();
+    }
+  } catch (err) {
+    console.error(
```

Comment:
What kind of error cases can be caught here? Is it only actual errors that represent bugs, or can the user also cause errors here by incorrectly using the software?

Reply:
I believe it is both. E.g. a user could send a request without `Authorization` header. Then `credentials === undefined` and calling `credentials.name` would raise an exception.
But yes, by just using the software (correctly or incorrectly) one should not run into this.

Reply:
If this is the case, I think we should differentiate the cases. Problems resulting from users incorrectly using the software should not be logged on the error level.

Reply:
Agreed. Changed it such that user-errors are logged on log-level and unexpected errors are logged on error-level.